src/MatrixClientPeg.js

```diff
@@ -24,10 +24,6 @@ function deviceId() {
     // XXX: is Math.random()'s deterministicity a problem here?
     var id = Math.floor(Math.random()*16777215).toString(16);
     id = "W" + "000000".substring(id.length) + id;
-    if (localStorage) {
-        id = localStorage.getItem("mx_device_id") || id;
-        localStorage.setItem("mx_device_id", id);
-    }
     return id;
 }
```

Comment:
Maybe rename this to `generateDeviceId()` now it no longer gets 'the' device id?